tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -331,9 +333,8 @@ TEST_F(
   TensorView* tva_j = select(tva, 0, j);
   TensorView* tvc_j = select(tvc, 0, j);
   TensorView* tvc_locally_reduced_j =
-      matmul(tva_j, tvb); // ideally we should use the preallocated global
-                          // buffer tc_locally_reduced, but ExpressionEvaluator
-                          // do not support preallocated output buffer.
+      select(tvc_locally_reduced, 0, stream_index);
+  auto matmul = IrBuilder::create<MatmulOp>(tvc_locally_reduced_j, tva_j, tvb);
```

Comment:
Nice! I recommend changes in ReduceScatterBasedPipeliningHostIrImplementation come as a separate PR as well. It probably has to depend on HostIrExecutor changes.

Reply:
same, this change is introduced in https://github.com/NVIDIA/Fuser/pull/3171